calico/felix/profilerules.py

```diff
@@ -282,19 +290,16 @@ def _update_chains(self):
         self._iptables_updater.rewrite_chains(updates, deps, async=False)
 
 
-def extract_tags_from_profile(profile):
+def extract_tags_and_selectors_from_profile(profile):
     if profile is None:
         return set()
-    tags = set()
+    tags_and_sels = set()
     for in_or_out in ["inbound_rules", "outbound_rules"]:
         for rule in profile.get(in_or_out, []):
-            tags.update(extract_tags_from_rule(rule))
-    return tags
-
-
-def extract_tags_from_rule(rule):
-    return set(rule[key] for key in ["src_tag", "dst_tag"]
-               if key in rule and rule[key] is not None)
+            for key in ["src_tag", "dst_tag", "src_selector", "dst_selector"]:
+                if key in rule:
+                    tags_and_sels.add(rule[key])
```

Comment:
Good spot: `not None` is policed by the validation logic (it strips out None values) so that code couldn't be hit.

The real reason for removing the set comprehension is an intermediate revision of this code but I just tried to put it back and the formatting really gets out of hand with the extra keys to examine so I think I'd rather leave as-is.